#!/usr/bin/env python
PACKAGE = "teleoperation"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("master_min_x",     double_t, 0, "master_min_x parameter",    0.12, -10.0, 10.0)
gen.add("master_max_x",     double_t, 0, "master_max_x parameter",    0.20, -10.0, 10.0)
gen.add("master_min_y",     double_t, 0, "master_min_y parameter",   -0.26, -10.0, 10.0)
gen.add("master_max_y",     double_t, 0, "master_max_y parameter",    0.26, -10.0, 10.0)
gen.add("master_min_z",     double_t, 0, "master_min_z parameter",   -0.01, -10.0, 10.0)
gen.add("master_max_z",     double_t, 0, "master_max_z parameter",    0.33, -10.0, 10.0)
gen.add("master_min_yaw",   double_t, 0, "master_min_yaw parameter", -2.57, -10.0, 10.0)
gen.add("master_max_yaw",   double_t, 0, "master_max_yaw parameter",  2.55, -10.0, 10.0)

gen.add("master_joint_min_x",   double_t, 0, "master_joint_min_x parameter",    1.0,  -10.0, 10.0)
gen.add("master_joint_max_x",   double_t, 0, "master_joint_max_x parameter",    1.0,  -10.0, 10.0)
gen.add("master_joint_min_y",   double_t, 0, "master_joint_min_y parameter",    1.0,  -10.0, 10.0)
gen.add("master_joint_max_y",   double_t, 0, "master_joint_max_y parameter",    1.0,  -10.0, 10.0)
gen.add("master_joint_min_z",   double_t, 0, "master_joint_min_z parameter",    1.0,  -10.0, 10.0)
gen.add("master_joint_max_z",   double_t, 0, "master_joint_max_z parameter",    1.0,  -10.0, 10.0)
gen.add("master_joint_min_yaw", double_t, 0, "master_joint_min_yaw parameter", -2.57, -10.0, 10.0)
gen.add("master_joint_max_yaw", double_t, 0, "master_joint_max_yaw parameter",  2.55, -10.0, 10.0)

gen.add("slave_velocity_min_x",   double_t, 0, "slave_velocity_min_x parameter",   -0.2, -2.0, 2.0)
gen.add("slave_velocity_max_x",   double_t, 0, "slave_velocity_max_x parameter",    0.2, -2.0, 2.0)
gen.add("slave_velocity_min_y",   double_t, 0, "slave_velocity_min_y parameter",   -0.2, -2.0, 2.0)
gen.add("slave_velocity_max_y",   double_t, 0, "slave_velocity_max_y parameter",    0.2, -2.0, 2.0)
gen.add("slave_velocity_min_z",   double_t, 0, "slave_velocity_min_z parameter",   -0.2, -2.0, 2.0)
gen.add("slave_velocity_max_z",   double_t, 0, "slave_velocity_max_z parameter",    0.2, -2.0, 2.0)
gen.add("slave_velocity_min_yaw", double_t, 0, "slave_velocity_min_yaw parameter", -0.2, -2.0, 2.0)
gen.add("slave_velocity_max_yaw", double_t, 0, "slave_velocity_max_yaw parameter",  0.2, -2.0, 2.0)

exit(gen.generate(PACKAGE, "teleoperation", "HapticTeleop"))
